proposals/metadata.md

```diff
@@ -0,0 +1,88 @@
+<!--
+Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"). You may
+not use this file except in compliance with the License. A copy of the
+License is located at
+
+     http://aws.amazon.com/apache2.0/
+
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+-->
+
+### Container metadata service
+In order to solve the issues related to container metadata like: [#288](https://github.com/aws/amazon-ecs-agent/issues/288) and [#456](https://github.com/aws/amazon-ecs-agent/issues/456), we are considering adding a new feature "Container Metadata Service" similar to [EC2 Instance Metadata Service](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) in ECS agent to provide the container metadata.
+
+By adding this feature, you can query the information about the task, container and container instance from within the container or from the container instance by reading the metadata file for each container. The following information will be available from this feature:
+ * Information from docker:
+  * Container NetworkSettings
+  * Container ID from docker
+  * Container Name
+  * Image Name
+  * Image ID
+  * Port Mapping
+ * Information from ECS:
+  * Cluster Arn
+  * Container Instance Arn
+  * Task Arn
+
+ECS agent will create a file for each container under the directory `/var/lib/ecs/data/` and then mount the file to the container under a directory eg: `/ecs/metadata` with read only permission when the container is being created. Also an environment variable denoting the file path eg: `ECS_CONTAINER_METADATA_FILE` will be injected into the container. After the container is started, ECS agent will update the metadata of container into this file. Then the container can get its metada information by parsing the json file with path `ECS_CONTAINER_METADATA_FILE`. This directory will be removed when task is being clean up. The detailed steps when start a container will be:
+* Create metadata file for each container under `/var/lib/ecs/data/` for each container.
+* Create the container with the two extra steps:
+  * Mount the metadata file created above to a specific directory eg: `/ecs/metadata/` with read only permission for each container.
+  * Inject the metadata file path as an environment variable to the container eg: `ECS_CONTAINER_METADATA_FILE=/ecs/metadata/container1`.
```

Comment:
I think we will pick up the container name from task definition, so the `ECS_CONTAINER_METADATA_FILE` is consisted of `/path/to/directory/${CONTAINER_NAME}`.
Will update the file to make it more clear, thanks.